frontend/packages/noobaa-storage-plugin/src/components/activity-card/activity-card.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import DashboardCard from '@console/shared/src/components/dashboard/dashboard-card/DashboardCard';
+import DashboardCardBody from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardBody';
+import DashboardCardHeader from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardHeader';
+import DashboardCardTitle from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardTitle';
+import { EventKind } from '@console/internal/module/k8s';
+import { FirehoseResource, FirehoseResult } from '@console/internal/components/utils';
+import { EventModel } from '@console/internal/models';
+import ActivityBody, {
+  RecentEventsBody,
+  OngoingActivityBody,
+} from '@console/shared/src/components/dashboard/activity-card/ActivityBody';
+import { PrometheusResponse } from '@console/internal/components/graphs';
+import {
+  DashboardItemProps,
+  withDashboardResources,
+} from '@console/internal/components/dashboard/with-dashboard-resources';
+import { DATA_RESILIENCE_QUERIES } from '../../queries';
+import { NooBaaObjectBucketClaimModel, NooBaaObjectBucketModel } from '../../models';
+import { isDataResiliencyActivity } from '../../utils';
+
+const eventsResource: FirehoseResource = { isList: true, kind: EventModel.kind, prop: 'events' };
+
+const ocsEventNamespaceKindFilter = (event: EventKind): boolean =>
+  _.get(event, 'involvedObject.kind') ===
+  (NooBaaObjectBucketClaimModel.kind || NooBaaObjectBucketModel.kind);
+
+const RecentEvent = withDashboardResources(
+  ({ watchK8sResource, stopWatchK8sResource, resources }: DashboardItemProps) => {
+    React.useEffect(() => {
+      watchK8sResource(eventsResource);
+      return () => {
+        stopWatchK8sResource(eventsResource);
+      };
+    }, [watchK8sResource, stopWatchK8sResource]);
+    return (
+      <RecentEventsBody
+        events={resources.events as FirehoseResult<EventKind[]>}
+        filter={ocsEventNamespaceKindFilter}
+      />
+    );
+  },
+);
+
+const OngoingActivity = withDashboardResources(
+  ({ watchPrometheus, stopWatchPrometheusQuery, prometheusResults }: DashboardItemProps) => {
+    React.useEffect(() => {
+      watchPrometheus(DATA_RESILIENCE_QUERIES.REBUILD_PROGRESS_QUERY);
+      return () => stopWatchPrometheusQuery(DATA_RESILIENCE_QUERIES.REBUILD_PROGRESS_QUERY);
+    }, [watchPrometheus, stopWatchPrometheusQuery]);
+
+    const progress = prometheusResults.getIn([
+      DATA_RESILIENCE_QUERIES.REBUILD_PROGRESS_QUERY,
+      'data',
+    ]) as PrometheusResponse;
+    const progressError = prometheusResults.getIn([
+      DATA_RESILIENCE_QUERIES.REBUILD_PROGRESS_QUERY,
+      'loadError',
+    ]);
+
+    const eta = prometheusResults.getIn([
+      DATA_RESILIENCE_QUERIES.REBUILD_TIME_QUERY,
+      'data',
```

Comment:
I think we should check for loadError for `REBUILD_TIME_QUERY`

Reply:
Even if the eta is loaded or not, we need to show the progress.